api/envoy/type/metadata/v2/metadata.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package envoy.type.metadata.v2;
+
+option java_outer_classname = "MetadataProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type.metadata.v2";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Metadata]
+
+// MetadataLocate provides a general interface using `filter` and `path` to retrieve value from
```

Comment:
I think I would call this `MetadataLookup` personally, WDYT?

@htuch should this live in type or somewhere else?

Reply:
what if we only place a general struct lookup here?

Reply:
I actually prefer `MetadataKey`; this is specifying a structured index into metadata. Also, we should tag `core.Metadata` as `[#next-major-version: move to type/metadata/v2]`.

Reply:
`MetadataKey` is fine with me.